nemo/collections/tts/data/datalayers.py

```diff
@@ -207,3 +217,223 @@ def __getitem__(self, index):
 
     def __len__(self):
         return self.samples.shape[0] // self.n_segments
+
+
+"""
+    This dataset was implmented using code from: https://github.com/Sytronik/deep-griffinlim-iteration/blob/master/dataset.py 
+"""
+
+
+class NoisySpecsDataset(Dataset):
+    @property
+    def output_types(self) -> Optional[Dict[str, NeuralType]]:
+        """Returns definitions of module output ports.
+               """
+        return {
+            'x': NeuralType(('B', 'C', 'F', 'T'), SpectrogramType()),
+            'mag': NeuralType(('B', 'any', 'F', 'T'), SpectrogramType()),
+            'max_length': NeuralType(None, LengthsType()),
+            'y': NeuralType(('B', 'C', 'F', 'T'), SpectrogramType()),
+            'T_ys': NeuralType(tuple('B'), LengthsType()),
+            'length': NeuralType(tuple('B'), LengthsType()),
+            'path_speech': NeuralType(tuple('B'), StringType()),
+        }
+
+    def __init__(
+        self, destination: Union[str, 'pathlib.Path'], subdir: str, n_fft: int, hop_length: int, num_snr: int,
+    ):
+        self.tar_dir = Path("%s/degli_data_%d_%dx%d/%s/" % (destination, n_fft, hop_length, num_snr, subdir))
+        """
+        A modified dataset for training deep-griffin-lim iteration. Contains MSTFT (mag), STFT (y) , and noisy STFT which is
+        used for initial phase. By using different levels of noise, the Degli model can learn to improve any phase, and thus
+        it can be used iteratively.  
+
+        Args:
+            destination (str, Path): Path to a directory containing the main data set folder, Similar to the directory
+            provided to the preprocessor script, which generates this dataset. 
+            subdir (str): Either 'train', or 'valid', when using the standard script for generation.
+            n_fft (int): STFT parameter. Also detrmines the STFT filter length.
+            hop_length (int): STFT parameter.
+            num_snr (int): number of noisy samples per clean audio in the original dataset.
+        """

```

Comment:
Your docstring mentions adding noise, but I don't see this anywhere. Can you point out where the noise is added?
At a quick glance, all it does is load data from a numpy file.